tic_tac_toe/src/board.py

```diff
@@ -0,0 +1,185 @@
+
+        x_or_o : string
+        which player we are checking for
+
+        returns
+        ----------
+        winner : boolean
+        whether or not that player won with a column win
+        """
+        board_width = self.dim
+        winner = None
+
+        for i in range(0, board_width):
+            values = set()
+            for j in range(0, board_width):
+                index = self.convert(i, j)
+                if self.values[index] != None:
+                    values.add(self.values[index])
+            if len(values) == 1:
+                if x_or_o == list(values)[0]:
+                    winner = True
+            else:
+                winner = False
+        return winner
+
+    def _check_row(self, x_or_o):
+        """
+        Checks if a specific player has won with a row win
+        Parameters
+        ----------
+        board : class
+        The board to operate on
+
+        x_or_o : string
+        Which player we are checking for
+
+        Returns
+        ----------
+        winner : boolean
+        Whether or not that player won with a row win
+        """
+        board_width = self.dim
+        winner = None
+        for i in range(0, board_width):
+            values = set()
+            for j in range(0, board_width):
+                index = self.convert(j, i)
+                if self.values[index] != None:
+                    values.add(self.values[index])
+            if len(values) == 1:
+                if x_or_o == list(values)[0]:
+                    winner = True
+            else:
+                winner = False
+        return winner
+
+    def _check_diagonal(self, x_or_o):
+        """
+        Checks if a specific player has won with a diagonal win
+        Parameters
+        ----------
+        board : class
+        The board to operate on
+
+        x_or_o : string
+        Which player we are checking for
+        Returns
+        ----------
+        winner : boolean
+        Whether or not that player won with a diagonal win
+        """
+
+        # Check of someone has won
+        board_width = self.dim
+        winner = None
+        values = set()
+        values2 = set()
+        for i in range(0, board_width):
+            index = self.convert(i, i)
+            index2 = self.convert(i, (-i + 2))
+            values.add(self.values[index])
+            values2.add(self.values[index2])
+        if len(values) == 1:
+            if x_or_o == list(values)[0]:
+                winner = True
+                return winner
+            else:
+                winner = False
+        else:
+            winner = False
+
+        if len(values2) == 1 and winner == False:
+            if x_or_o == list(values2)[0]:
+                winner = True
+                return winner
+            else:
+                winner = False
+        else:
+            winner = False
+        return winner
+
+    def check_winner(self, x_or_o, tof):
+        if self._check_column(x_or_o) == tof:
```

Comment:
Why not just `self._check_column(x_or_o) or self._check_row(x_or_o) or self.check_diagonal(x_or_o)`?

Reply:
What is `tof` here?  Don't all of these `_check_*` functions return booleans?  Can you add a docstring here?